### PR TITLE
Yet another tranche of small fixes

### LINKED ISSUE #112: Transceiver.java ExecuteLock
_Remember: the better you describe your issue, the easier it is for us to help you._

What is the problem you see?
the class def for ExecuteLock is at lines 676 to 698, the usage of said class doesnt start till 1291.
============================

What do you see that is wrong?
the executeLock is nowhere near the code that uses it

What do you think it should do instead?
either:
1. be located next to where its used
2. plonk all these extra classes at the bottom of the class, so that we're not muddying the water

I personally perfer 2. 

What version of our software are you using?
master
-------------------------------------------

### LINKED ISSUE #167: APP_MASK
_Remember: the better you describe your issue, the easier it is for us to help you._

What is the problem you see?
constant cloned in many places.
============================

What do you see that is wrong?
should be in 1 place and referred

What do you think it should do instead?

What version of our software are you using?
master
-------------------------------------------

### LINKED ISSUE #168: ALL_CORE_MASK
_Remember: the better you describe your issue, the easier it is for us to help you._

What is the problem you see?
cloned in 2 places (sendSignal, and CountState)
============================

What do you see that is wrong?
one place

What do you think it should do instead?

What version of our software are you using?
master
-------------------------------------------

### LINKED ISSUE #169: ApplicationStop.java line 43
_Remember: the better you describe your issue, the easier it is for us to help you._

What is the problem you see?
	private static int argument2(int appID, Signal signal) {
============================

What do you see that is wrong?
the ide states that signal is always Signal.STOP

What do you think it should do instead?
well assuming its right, which given its a SCPRequest and is a argument2 generator might mean it can be fixed

What version of our software are you using?
master
-------------------------------------------

### LINKED ISSUE #200: camel case
Full path to where you find the problem:
C:\Users\alan\Documents\software\JavaSpiNNaker\SpiNNaker-utils\src\main\java\uk\ac\manchester\spinnaker\utils\RawConfigParser.java

What is the problem you see?
public Integer getint(String section, String option) {
public Boolean getboolean(String section, String option) {

What do you think it should do instead?
public Integer getInt(String section, String option) {
public Boolean getBoolean(String section, String option) {

What version of our software are you using?
master

### LINKED ISSUE #201: lambda vs method reference
Full path to where you find the problem:
C:\Users\alan\Documents\software\JavaSpiNNaker\SpiNNaker-comms\src\main\java\uk\ac\manchester\spinnaker\connections\SCPRequestPipeline.java

What is the problem you see?
		private boolean allTimeoutFailures() {
			return retryReason.stream().allMatch(r -> REASON_TIMEOUT.equals(r));
		}

What do you think it should do instead?
this seems to be flagged, due to the changes of the constant instead of param
return retryReason.stream().allMatch(REASON_TIMEOUT::equals);

What version of our software are you using?
more-fixing

### LINKED ISSUE #202: redundant param
Full path to where you find the problem:
C:\Users\alan\Documents\software\JavaSpiNNaker\SpiNNaker-comms\src\main\java\uk\ac\manchester\spinnaker\connections\UDPConnection.java

What is the problem you see?
InetSocketAddress addr = new InetSocketAddress(localAddr, localPort);
		return addr;

What do you think it should do instead?
return  new InetSocketAddress(localAddr, localPort);

What version of our software are you using?
more-fixing

### LINKED ISSUE #203: redundant throw
Full path to where you find the problem:
C:\Users\alan\Documents\software\JavaSpiNNaker\SpiNNaker-comms\src\main\java\uk\ac\manchester\spinnaker\connections\UDPConnection.java

What is the problem you see?
@Override
	public boolean isConnected() throws IOException {
		if (!canSend) {
			return false;
		}
		for (int i = 0; i < PING_COUNT; i++) {
			if (ping(remoteIPAddress) == 0) {
				return true;
			}
		}
		return false;
	}

What do you think it should do instead?
IOException is redundant. go bye bye

What version of our software are you using?
more-fixing

### LINKED ISSUE #206: defaultChip vs BootChip
Full path to where you find the problem:
starts in SCPRequest (C:\Users\alan\Documents\software\JavaSpiNNaker\SpiNNaker-comms\src\main\java\uk\ac\manchester\spinnaker\messages\scp\SCPRequest.java)

What is the problem you see?
	public static final CoreLocation DEFAULT_MONITOR_CORE =
			new CoreLocation(DEFAULT_DEST_X, DEFAULT_DEST_Y, 0);
    /**
	 * The pseudo-location of the default chip.
	 */
	public static final ChipLocation DEFAULT_CHIP =
			DEFAULT_MONITOR_CORE.asChipLocation();

What do you think it should do instead?
the 255 255 is actually the boot chip, not the default chip. took a good while with @lplana and @rowleya to get to that logic, but basically, default_chip is a misnomer

What version of our software are you using?
more-fixing

### PR DESCRIPTION
**Pull request by dkfellows**
fixes #112 
fixes #167 
fixes #168 
fixes #169 
fixes #200 
fixes #201 
fixes #202
fixes #203 
fixes #206

### CODE REVIEW

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/ApplicationRun.java [hunk 1]
```diff
@@ -31,16 +32,20 @@
...
 	 *            The processors of the chip to run on, between 1 and 17
 	 */
-	public ApplicationRun(int appId, HasChipLocation chip,
+	public ApplicationRun(int appID, HasChipLocation chip,
 			Iterable<Integer> processors) {
-		this(appId, chip, processors, false);
+		this(appID, chip, processors, false);
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/ApplicationRun.java]: 255 should use the MAX_APP_ID here. else risk of error message falling out of date over time

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/ApplicationStop.java [hunk 2]
```diff
@@ -54,8 +53,12 @@ private static int argument3() {
 	 *            The ID of the application, between 0 and 255
 	 */
 	public ApplicationStop(int appID) {
-		super(DEFAULT_MONITOR_CORE, CMD_NNP, argument1(),
-				argument2(appID, STOP), argument3());
+		super(BOOT_MONITOR_CORE, CMD_NNP, argument1(), argument2(appID),
+				argument3());
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/ApplicationStop.java]: same issue here

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/CountState.java [hunk 3]
```diff
@@ -29,22 +32,24 @@
...
 	 */
-	public CountState(int appId, CPUState state) {
-		super(DEFAULT_MONITOR_CORE, CMD_SIG, POINT_TO_POINT.value,
-				argument2(appId, state), ALL_CORE_MASK);
+	public CountState(int appID, CPUState state) {
+		super(BOOT_MONITOR_CORE, CMD_SIG, POINT_TO_POINT.value,
+				argument2(appID, state), ALL_CORE_SIGNAL_MASK);
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/CountState.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/FixedRouteInitialise.java [hunk 4]
```diff
@@ -39,6 +40,10 @@ private static int argument1(int appID) {
 	 */
 	public FixedRouteInitialise(HasChipLocation chip, int entry, int appID) {
 		super(chip.getScampCore(), CMD_RTR, argument1(appID), entry);
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/FixedRouteInitialise.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/FixedRouteRead.java [hunk 5]
```diff
@@ -48,6 +49,10 @@ private static int argument2() {
 	 */
 	public FixedRouteRead(HasChipLocation chip, int appID) {
 		super(chip.getScampCore(), CMD_RTR, argument1(appID), argument2());
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/FixedRouteRead.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/RouterAlloc.java [hunk 6]
```diff
@@ -43,6 +44,10 @@
 	public RouterAlloc(HasChipLocation chip, int appID, int numEntries) {
 		super(chip.getScampCore(), CMD_ALLOC, argument1(appID), numEntries);
 		this.numEntries = numEntries;
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/RouterAlloc.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/RouterInit.java [hunk 7]
```diff
@@ -43,6 +44,10 @@ public RouterInit(HasChipLocation chip, int numEntries, int tableAddress,
 			int baseAddress, int appID) {
 		super(chip.getScampCore(), CMD_RTR, argument1(numEntries, appID),
 				tableAddress, baseAddress);
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/RouterInit.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SDRAMAlloc.java [hunk 8]
```diff
@@ -59,6 +60,10 @@ public SDRAMAlloc(HasChipLocation chip, int appID, int size) {
 	public SDRAMAlloc(HasChipLocation chip, int appID, int size, int tag) {
 		super(chip.getScampCore(), CMD_ALLOC, argument1(appID), size, tag);
 		this.size = size;
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SDRAMAlloc.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SDRAMDeAlloc.java [hunk 9]
```diff
@@ -58,6 +59,10 @@ public SDRAMDeAlloc(HasChipLocation chip, int appID) {
 	public SDRAMDeAlloc(HasChipLocation chip, int appID, int baseAddress) {
 		super(chip.getScampCore(), CMD_ALLOC, (int) FREE_SDRAM_BY_POINTER.value,
 				baseAddress);
+		if (appID < 0 || appID > MAX_APP_ID) {
+			throw new IllegalArgumentException(
+					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SDRAMDeAlloc.java]: same

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SendSignal.java [hunk 10]
```diff
@@ -27,19 +30,15 @@
...
 	 *            The coordinates of the chip to run on
 	 */
 	public SendSignal(int appID, Signal signal) {
-		super(DEFAULT_MONITOR_CORE, CMD_SIG, signal.type.value,
-				argument2(appID, signal), ALL_CORE_MASK);
+		super(BOOT_MONITOR_CORE, CMD_SIG, signal.type.value,
+				argument2(appID, signal), ALL_CORE_SIGNAL_MASK);
 		if (appID < 0 || appID > MAX_APP_ID) {
 			throw new IllegalArgumentException(
 					"appID must be between 0 and 255");
```
Comment (alan-stokes) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/messages/scp/SendSignal.java]: same

**Comment by dkfellows** [comment]
I think I want to make `appID` into something other than effectively a bare unsigned byte…

**Comment by alan-stokes** [comment]
such as?

**Comment by dkfellows** [comment]
Such as the commit I've just done.

**Comment by alan-stokes** [comment]
im happy with that

**Pull request merged**